pydatastructs/trees/binary_trees.py

```diff
@@ -1057,6 +1047,357 @@ def split(self, x):
             self.tree[self.root_idx].right = None
         return other
 
+class Redblacktree(SelfBalancingBinaryTree):
+    """
+    Represents Red Black trees.
+
+    Examples
+    ========
+
+    >>> from pydatastructs.trees import Redblacktree as RB
+    >>> b = RB()
+    >>> b.insert(1,1)
+    >>> b.insert(2,2)
+    >>> child = b.tree[b.root_idx].right
+    >>> b.tree[child].data
+    2
+    >>> b.search(1)
+    0
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Red%E2%80%93black_tree
+
+    See Also
+    ========
+
+    pydatastructs.trees.binary_tree.SelfBalancingBinaryTree
+    """
+
+    @classmethod
+    def methods(cls):
+        return ['__str__', 'insert', 'delete']
+
+    def _get_parent(self,node_idx):
+        return self.tree[node_idx].parent
+
+    def _get_grand_parent(self,node_idx):
+        parent_idx=self._get_parent(node_idx)
+        return self.tree[parent_idx].parent
+
+    def _get_sibling(self,node_idx):
+        parent_idx=self._get_parent(node_idx)
+        if parent_idx is None:
+            return None
+        node = self.tree[parent_idx]
+        if node_idx==node.left:
+            sibling_idx=node.right
+            return sibling_idx
+        else:
+            sibling_idx=node.left
+            return sibling_idx
+
+    def _get_uncle(self,node_idx):
```

Comment:
```suggestion
    def _get_uncle(self, node_idx):
```
Please use a space after, `,` at every other place in your patch.